packages/web_benchmarks/lib/src/benchmark_result.dart

```diff
@@ -31,11 +35,18 @@ class BenchmarkScore {
   /// The result of measuring a particular metric in this benchmark run.
   final num value;
 
+  /// Optional delta value describing the difference between this metric's score
+  /// and the score of a matching metric from another [BenchmarkResults].
+  ///
+  /// This value may be assigned by the [computeDelta] analysis method.
+  num? delta;
```

Comment:
This class is designed to be immutable. Is it possible to instead make this field `final`, add it to the constructor as an optional field, and then have `computeDelta` create new `BenchmarkScore` objects (and a new `BenchmarkResults` object) that contain the deltas, while keeping the baseline and test object intact?

Reply:
done